src/components/infinite-scroll/__tests__/InfiniteScroll.test.tsx

```diff
@@ -16,154 +13,108 @@ const propsList: InfiniteScrollProps = {
-
-        window.dispatchEvent(new Event('scroll'));
-        expect(mockOnLoadMore).toHaveBeenCalledTimes(1);
-    });
+    describe('using window', () => {
+        beforeEach(() => {
+            component = mount(
+                <InfiniteScroll {...propsList} hasMore>
+                    <div>
+                        {items.map((item, i) => (
+                            <div key={i} style={{ height: '100px' }}>
+                                {item}
+                            </div>
+                        ))}
+                    </div>
+                </InfiniteScroll>,
+                { attachTo },
+            );
+        });
 
-    it('should not call onLoadMore if sentinel is not in threshold range while scrolling in window', () => {
-        const items = new Array(20).fill('ITEM');
-        mount(
-            <InfiniteScroll {...propsList} hasMore>
-                <div>
-                    {items.map((item, i) => (
-                        <div key={i}>{item}</div>
-                    ))}
-                </div>
-            </InfiniteScroll>,
-            { attachTo },
-        );
-
-        const sentinel = getSentinel();
-        sandbox
-            .stub(sentinel, 'getBoundingClientRect')
-            .returns({ top: window.innerHeight + (threshold + 1) } as DOMRect);
-
-        window.dispatchEvent(new Event('scroll'));
-        expect(mockOnLoadMore).not.toHaveBeenCalled();
-    });
+        it('should call onLoadMore if sentinel is in threshold range and window is not scrollable', () => {
+            const sentinel = getSentinel();
+            sentinel.getBoundingClientRect = jest
+                .fn()
+                .mockReturnValue({ top: window.innerHeight + (threshold - 1) } as DOMRect);
 
-    it('should call onLoadMore if sentinel is in threshold range while scrolling scrollContainerNode', () => {
-        const scrollContainer = document.createElement('div');
-        sandbox.stub(scrollContainer, 'getBoundingClientRect').returns({ bottom: 500 } as DOMRect);
-
-        const items = new Array(20).fill('ITEM');
-        mount(
-            <InfiniteScroll {...propsList} hasMore scrollContainerNode={scrollContainer} useWindow={false}>
-                <div>
-                    {items.map((item, i) => (
-                        <div key={i} style={{ height: '100px' }}>
-                            {item}
-                        </div>
-                    ))}
-                </div>
-            </InfiniteScroll>,
-            { attachTo },
-        );
-
-        const sentinel = getSentinel();
-        sandbox.stub(sentinel, 'getBoundingClientRect').returns({ top: 500 + (threshold - 1) } as DOMRect);
-
-        scrollContainer.dispatchEvent(new Event('scroll'));
-        expect(mockOnLoadMore).toHaveBeenCalledTimes(1);
-    });
+            // update prop to trigger useEffect
+            component.setProps({ throttle: 1 });
+            expect(mockOnLoadMore).toHaveBeenCalledTimes(1);
```

Comment:
could we change this to `toBeCalledTimes(1)`?

Reply:
as well as the others from `toHaveBeen[...]` to `toBe[...]`

Reply:
yup! didn't notice as I was modifying the original tests